blockdag/dag.go

```diff
@@ -1153,31 +1131,35 @@ func (node *blockNode) fetchBlueBlocks() ([]*util.Block, error) {
 // applyBlueBlocks adds all transactions in the blue blocks to the selectedParent's UTXO set
 // Purposefully ignoring failures - these are just unaccepted transactions
 // Writing down which transactions were accepted or not in txsAcceptanceData
-func (node *blockNode) applyBlueBlocks(acceptedSelectedParentUTXO UTXOSet, selectedParentAcceptanceData []TxAcceptanceData, blueBlocks []*util.Block) (
+func (node *blockNode) applyBlueBlocks(selectedParentPastUTXO UTXOSet, blueBlocks []*util.Block) (
 	pastUTXO UTXOSet, multiBlockTxsAcceptanceData MultiBlockTxsAcceptanceData, err error) {
 
-	pastUTXO = acceptedSelectedParentUTXO
-	multiBlockTxsAcceptanceData = MultiBlockTxsAcceptanceData{BlockTxsAcceptanceData{
-		BlockHash:        *node.selectedParent.hash,
-		TxAcceptanceData: selectedParentAcceptanceData,
-	}}
+	selectedParentPastDiffUTXOSet, ok := selectedParentPastUTXO.(*DiffUTXOSet)
+	if !ok {
+		panic("selectedParentPastUTXO is not *DiffUTXOSet")
+	}
+	pastUTXO = NewDiffUTXOSet(selectedParentPastDiffUTXOSet.base, selectedParentPastDiffUTXOSet.UTXODiff.clone())
```

Comment:
Why not use utxoSet.clone() for this whole thing?
No errors, no panics, no type assertions.

Reply:
Note: right now DiffUTXOSet.clone actually clones the base as well.
I'm not sure why is it useful, so I suggest changing this behavior. (Ofc, make sure it makes sense in all usages)